### PR TITLE
DEV2-2728: Add authentication

### PR DESCRIPTION
**Pull request by ofekby**

### CODE REVIEW

File: charts/tabnine-cloud/values.yaml [hunk 1]
```diff
@@ -170,6 +170,7 @@ server:
 frontend:
   enabled: true
   enforceJWT: true
+  enforceUserInTeamAuthentication: true
```
Comment (yanir-codota) [inline charts/tabnine-cloud/values.yaml:173]: we discussed removing this one, no?
what was the decision?

Comment (ofekby) [reply charts/tabnine-cloud/values.yaml:173]: Not this, the license key flag was removed, this controls weather the user must be in a team or just authenticated

File: charts/tabnine-cloud/templates/frontend/deployment.yaml [hunk 2]
```diff
@@ -44,8 +44,10 @@ spec:
           env:
           - name: TRITON_SERVER
             value: http://{{ include "tabnine-cloud.severName" . }}.{{ .Release.Namespace}}:{{ .Values.server.service.port }}
-          - name: AUTH_SERVER_BASE_URL 
+          - name: AUTH_SERVER_BASE_URL
             value: http://{{ include "tabnine-cloud.fullname" . }}-auth.{{ .Release.Namespace}}:{{ .Values.auth.service.port }}
+          - name: ENFORCE_USER_IN_TEAM_AUTHENTICATION
+            value: {{ .Values.frontend.enforceUserInTeamAuthentication }}
```
Comment (assapir) [inline charts/tabnine-cloud/templates/frontend/deployment.yaml:50]: you will probably need to have `"` around, as env vars are always string

File: charts/tabnine-cloud/values.yaml [hunk 3]
```diff
@@ -177,6 +177,7 @@ server:
...
```
Comment (assapir) [inline charts/tabnine-cloud/values.yaml:180]: suggest to move that to `global.tabnine` - as it general config that just impl in the frontend (also, do we need `forceJWT` if that is on?)

**Review by yanir-codota — APPROVED** [review]
LGTM

File: charts/tabnine-cloud/values.yaml (hunk 3 above)
Comment (ofekby) [reply charts/tabnine-cloud/values.yaml:193]: I think it will be better if we heep it here, I see it as a FF and I think we should remove it in a time.

**Pull request merged**